source/core/serial_solid.cc

```diff
@@ -45,10 +44,15 @@ SerialSolid<dim, spacedim>::SerialSolid(
   , param(param)
   , id(id)
   , output_bool(param->output_bool)
-  , translational_velocity(&param->translational_velocity)
-  , angular_velocity(&param->angular_velocity)
   , center_of_rotation(param->center_of_rotation)
 {
+  translational_velocity =
+    std::make_shared<Functions::ParsedFunction<spacedim>>(spacedim);
+  angular_velocity =
+    std::make_shared<Functions::ParsedFunction<spacedim>>(spacedim);
+  translational_velocity = param->translational_velocity;
+  angular_velocity       = param->angular_velocity;
```

Comment:
If those are shared point (translational velocity and angular velocity) you don't need to allocate them. You just set them to be equal to the other ones because they should point to the same functions
Otherwise you can call the copy constructor, but there is no need to allocate to then erase the pointer.